src/app/components/main/main.component.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, OnInit} from '@angular/core';
+import {MoviesService} from "../../services/movies/movies.service";
+import {Movie} from "../../model/movie/movie";
+import {FindMovieService} from "../../services/findMovie/find-movie.service";
+
+@Component({
+  selector: 'app-main',
+  templateUrl: './main.component.html',
+  styleUrls: ['./main.component.scss']
+})
+export class MainComponent implements OnInit {
+  public movies: Movie[] = [];
+  public moviesToShow: Movie[] = [];
+  private movieToFind: string = "";
+
+  constructor(private moviesService: MoviesService, private findService: FindMovieService) {
+  }
+
+  ngOnInit(): void {
+    this.moviesService.moviesDtoObs
+      .subscribe(value => {
+        this.movies = value.data;
+        this.moviesToShow = this.movies;
+      });
+
+    this.findService.subscribe((movie: string) => {
+      this.moviesToShow = this.moviesToShow.filter(e => e.title === movie);
+    });
+
+  }
+
+  filterMovies(event: string): void {
+    if (event === "All") {
+      this.moviesToShow = this.movies;
+      return
+    }
+    this.moviesToShow = this.movies.filter(e => e.genres.indexOf(event) >= 0);
```

Comment:
I would add the filter and sorting logic into the moviesService. The service can expose just movies as an observable, which initially contains all movies, but when you filter or sort or add new movies, the movies will be updated. So this component just receives the movies to display and will use the moviesService api to filter/sort/add.